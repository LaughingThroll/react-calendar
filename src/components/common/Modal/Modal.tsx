import React from 'react'

import { TChildren } from '../../../types/utilsTypes'

interface IModalProps {
  open?: boolean
  children?: TChildren
  onClose?: () => void
}

const Modal: React.FC<IModalProps> = ({ children, open, onClose }) => {
  const handleClick = (e: React.MouseEvent) => {
    const target = e.target as HTMLElement
    if (!target.closest('.modal') && onClose) {
      onClose()
    }
  }

  return (
    <>
      {open && (
        <div className="overlay" onClick={handleClick}>
          <div className="modal">{children}</div>
        </div>
      )}
    </>
  )
}

export default Modal
