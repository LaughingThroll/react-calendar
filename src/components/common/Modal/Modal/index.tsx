import React from 'react'
import styles from './modal.module.scss'
import { TChildren } from '../../../../types/utilsTypes'

export interface ModalProps {
  open: boolean
  onClose: () => void
  children?: TChildren
}

const Modal: React.FC<ModalProps> = ({ children, open, onClose }) => {
  const { overlay, modal } = styles

  const handleClick = (e: React.MouseEvent) => {
    const target = e.target as HTMLDivElement
    if (target?.id === 'overlay') onClose()
  }

  return (
    <>
      {open && (
        <div id="overlay" className={overlay} onClick={handleClick}>
          <div className={modal}>{children}</div>
        </div>
      )}
    </>
  )
}

export default React.memo(Modal)
