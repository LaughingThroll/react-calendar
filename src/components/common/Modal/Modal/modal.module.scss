@import 'styles/utils/variables';

.overlay {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-color: $overlay;
  z-index: 100;
}

.modal {
  min-width: 400px;
  border-radius: 5px;
  overflow: hidden;
  background-color: $modal-bg;
}
