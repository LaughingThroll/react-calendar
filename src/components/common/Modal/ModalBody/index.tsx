import React from 'react'
import styles from './modalBody.module.scss'

interface ModalBodyProps {
  children?: any
}
const { modalBody, modalBody__item } = styles

const ModalBody: React.FC<ModalBodyProps> = ({ children }) => {
  return (
    <form className={modalBody}>
      {React.Children.map(children, (child, i) => (
        <div key={i} className={modalBody__item}>
          {child}
        </div>
      ))}
    </form>
  )
}

export default React.memo(ModalBody)
