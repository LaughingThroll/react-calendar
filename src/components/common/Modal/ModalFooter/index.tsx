import React from 'react'
import styles from './modalFooter.module.scss'

interface ModalFooterProps {
  children?: JSX.Element | JSX.Element[]
  className?: string[]
}

const { modalFooter } = styles

const ModalFooter: React.FC<ModalFooterProps> = ({ children, className = [] }) => (
  <div className={`${modalFooter} ${className.join(' ')}`}>{children}</div>
)

export default React.memo(ModalFooter)
