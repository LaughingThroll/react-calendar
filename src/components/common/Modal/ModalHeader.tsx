import React from 'react'

interface IModalHeader {
  className?: string[]
}

const ModalHeader: React.FC<IModalHeader> = ({ className = [], children }) => {
  return <div className={`modal-header ${className.join(' ')}`}>{children}</div>
}

export default ModalHeader
