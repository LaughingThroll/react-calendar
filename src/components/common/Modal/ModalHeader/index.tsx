import React from 'react'
import styles from './modalHeader.module.scss'
import { TChildren } from '../../../../types/utilsTypes'

interface ModalHeaderProps {
  className?: string[]
  children?: TChildren
}

const { modalHeader } = styles

const ModalHeader: React.FC<ModalHeaderProps> = ({ className = [], children }) => {
  return <div className={`${modalHeader} ${className.join(' ')}`}>{children}</div>
}

export default React.memo(ModalHeader)
