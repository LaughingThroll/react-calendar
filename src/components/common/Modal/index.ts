export { default as Modal } from './Modal/index'
export { default as ModalHeader } from './ModalHeader/index'
export { default as ModalBody } from './ModalBody/index'
export { default as ModalFooter } from './ModalFooter/index'
