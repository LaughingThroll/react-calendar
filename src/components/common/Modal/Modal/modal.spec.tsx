import React from 'react'
import { shallow } from 'enzyme'
import Modal, { ModalProps } from './'

const setUp = (props: ModalProps) => shallow(<Modal {...props} />)

describe('Modal render', () => {
  it('Modal default init', () => {
    const tree = setUp({ open: false, onClose: jest.fn() })

    expect(tree).toMatchSnapshot()
    expect(tree.find('.overlay')).toHaveLength(0)
    expect(tree.find('.modal')).toHaveLength(0)
  })

  it('Modal opened', () => {
    const tree = setUp({ open: true, onClose: jest.fn() })

    expect(tree).toMatchSnapshot()
    expect(tree.find('.overlay')).toHaveLength(1)
    expect(tree.find('.modal')).toHaveLength(1)
  })
})

describe('Modal prop onClose', () => {
  it('Modal onClose click true', () => {
    const props = { open: true, onClose: jest.fn() }
    const tree = setUp(props)
    const $overlay = tree.find('.overlay')
    const e = new Event('click')
    $overlay.simulate('click', [e])
    props.onClose()
    expect(props.onClose).toHaveBeenCalled()
  })
})
