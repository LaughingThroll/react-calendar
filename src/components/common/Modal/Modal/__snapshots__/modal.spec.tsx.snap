// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render Modal default init 1`] = `<Fragment />`;

exports[`Modal render Modal opened 1`] = `
<Fragment>
  <div
    className="overlay"
    id="overlay"
    onClick={[Function]}
  >
    <div
      className="modal"
    />
  </div>
</Fragment>
`;
